dolphinscheduler-python/pydolphinscheduler/src/pydolphinscheduler/tasks/spark.py

```diff
@@ -0,0 +1,110 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+"""Task Spark."""
+
+from typing import Dict, Optional
+
+from pydolphinscheduler.constants import TaskType
+from pydolphinscheduler.core.resource import Resource
+from pydolphinscheduler.core.task import Task
+
+
+class ProgramType(str):
```

Comment:
It seem that task spark and flink have some same membe, such as class `ProgramType`, `DeployMode` and even the way we get propetry `main_jar`. Should we add a parent class for task `spark` and `flink`? WDYT @devosend

Reply:
yes, `spark` and `flink` are similar, besides `mapreduce`. They have the same `ProgramType ` and `main_jar `. We should dd a parent class for them.

Reply:
So, what should be the name of the parent class? BigData? BigEngine? Engine? WDYT @zhongjiajie

Reply:
Oh, nameing things, always him 😭

Reply:
I do not want a name with too long, so I think `Engine` may be a good choice, but not quite that meaning too. cause we have SQL task type, which is also a `Engine` API which is more convenient

Reply:
Maybe we should naming it as `Engine` for temporary

Reply:
`DataCalculation` WDYT?

Reply:
It's too long for me.

Reply:
But…… I have no idea about that.

Reply:
I get it

Reply:
IMO `Engine` is better than `DataCalculation`